### PR TITLE
US16635 Project Page

### PR DESCRIPTION
**Pull request by tcmacdonald**
This PR implements the following UI element & business logic within the `<minimum-age-notification />` component...

screen shot 2019-03-07 at 4 41 38 pm

I'm also pulling in some of @CamiB's commits from [feature/US16635-ProjectPage](https://github.com/crdschurch/go-local/tree/feature/US16635-ProjectPage) that render an error message on 18+ projects. 

I abstracted one method from `<minimum-age-notification />` component into a new service, which is injected where needed throughout the app. 

Due to some upstream CORS issues, I _have not_ rebased with development yet so there are a few conflicts. Once those issues are resolved, let me know and I can resolve the conflicts so this can be merged easily. 

Let me know if you have any questions.

### CODE REVIEW

**Comment by vchtelmakh** [comment]
Please clear the hover-over behavior on the anchor tag displaying the min age; hovering over it and getting an underline implies that it's clickable to me, and I don't think it should be.

File: src/app/min-age-notification/min-age-notification.component.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+      &:after {
+        content:"";
+        position:absolute;
+        left: 0;
+        top: 0;
+        border-top:1px solid lighten($cr-gray-light, 10);
+        transform: rotate(45deg);
+        transform-origin: 0% 0%;
+        width: 51px;
```
Comment (vchtelmakh) [inline src/app/min-age-notification/min-age-notification.component.scss]: I had no idea how I would do the "cross-out" line if I had to - fancy!

**Comment by tcmacdonald** [comment]
> Please clear the hover-over behavior on the anchor tag displaying the min age; hovering over it and getting an underline implies that it's clickable to me, and I don't think it should be.

Good catch. Resolved in e09f3de

File: package.json [hunk 2]
```diff
@@ -11,9 +11,9 @@
     "start": "./node_modules/.bin/ng serve",
     "build-local": "./node_modules/.bin/ng build --aot true --build-optimizer --watch",
     "build": "./node_modules/.bin/ng build --configuration=$CRDS_ENV --output-hashing=none",
-    "test": "./node_modules/.bin/ng test --reporters teamcity --single-run --code-coverage",
+    "test": "./node_modules/.bin/ng test --reporters teamcity --watch=false --code-coverage",
     "test-watch": "./node_modules/.bin/ng test",
-    "test-coverage": "./node_modules/.bin/ng test --code-coverage --single-run",
+    "test-coverage": "./node_modules/.bin/ng test --code-coverage --watch=false",
```
Comment (vchtelmakh) [inline package.json]: It does hang atm, hopefully this fixes it.

File: src/app/min-age-notification/min-age-notification.component.html [hunk 3]
```diff
@@ -1,2 +1,7 @@
-<div id="minAgeNotification" [innerHtml]=minAgeNotificationText></div>
-
+<div class="min-age-notification font-family-condensed-extra">
+  <strong>Age Minimum</strong>
+  <a href="#" (click)="$event.preventDefault();" [ngClass]="{'on': projMinAge == 2}"><span><span>2+</span></span></a>
+  <a href="#" (click)="$event.preventDefault();" [hidden]="projMinAge < 8" [ngClass]="{'on': projMinAge == 8}"><span><span>8+</span></span></a>
+  <a href="#" (click)="$event.preventDefault();" [hidden]="projMinAge < 13" [ngClass]="{'on': projMinAge == 13}"><span><span>13+</span></span></a>
+  <a href="#" (click)="$event.preventDefault();" [hidden]="projMinAge < 18" [ngClass]="{'on': projMinAge == 18}"><span><span>18+</span></span></a>
```
Comment (vchtelmakh) [inline src/app/min-age-notification/min-age-notification.component.html]: 👍

File: src/app/services/date-time-formatter/date-time-formatter.service.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import * as moment from 'moment';
```
Comment (vchtelmakh) [inline src/app/services/date-time-formatter/date-time-formatter.service.ts]: We deleted this service entirely, it shouldn't be in the application at all.

File: src/styles.scss [hunk 5]
```diff
@@ -107,12 +108,38 @@ header.header {
...
 }
 
+.age-minimum-exception {
+  font-family: "lexia", serif;
+  font-style: italic;
+  color: red;
+  margin-top: 5px;
+}
+
+// .search-result {
```
Comment (vchtelmakh) [inline src/styles.scss]: I am fairly sure this comment was also deleted in an earlier PR. Looks like something happened during the merge.

**Pull request closed without merging**